File: src/main/java/com/syncstream_dev/syncstream/model/session/PublicSession.java
package com.syncstream_dev.syncstream.model.session;

import com.syncstream_dev.syncstream.model.ChatStream;
import com.syncstream_dev.syncstream.model.Genre;
import com.syncstream_dev.syncstream.model.VideoInfo;

public class PublicSession extends Session {
    public PublicSession(String sessionId, Genre genre, ChatStream chatStream, VideoInfo videoInfo) {
        super(sessionId, genre, chatStream, videoInfo);
    }
}

File: syncstream/src/datamodules/session/public_session.rs
// public_session.rs implements the PublicSession module

pub struct PublicSession {
    info: SessionInfo
}

impl PublicSession {
    pub fn new(
        id: Id,
        genre: Genre,
        participants: Vec<User>,
        chat_stream: Chatstream,
        video_info: VideoInfo,
    ) -> Self {
        let sessionInfo = SessionInfo::new(id, genre, participants, chat_stream, video_info);
        PublicSession {
            info: sessionInfo,
        }
    }
}
